ci/docker_build_r_sanitizer.sh

```diff
@@ -19,12 +19,14 @@
 set -e
 
 : ${R_BIN:=RDsan}
-source /arrow/ci/docker_install_r_deps.sh
 
 # Build arrow
 pushd /arrow/r
 
-install_deps
+# Install dependencies
+${R_BIN} -e "install.packages(c('remotes', 'dplyr', 'glue'))"
```

Comment:
This code will go out of date as soon as someone modify the `docker_build_r.sh` image but not this one. Either we merge scripts, or we pay close attention (I know I'll forget in k months).

Reply:
I'll add a comment in both scripts to remind us to update both if changes are needed in the future.

Fortunately, the `remotes::install_deps()` insulates us from most changes that could happen because that reads the dependency list in `r/DESCRIPTION`. 

Another reason not to merge scripts: does it make sense that I could use both conda and the R sanitizer build? I don't think it's possible, but logically you could make that happen with the right combination of env vars.